src/bg/updater.js

```diff
@@ -101,7 +101,8 @@ window._pickNextScriptAutoUpdate = async function() {
         let uuid = k.replace('updateNextAt.', '');
         let v = vs[k];
```

Comment:
Thanks so much for your effort!

After reading all the discussion and thinking about it, what do you think of making this line `let v = vs[k] || 0;`?  That's probably self-documenting.  And would still fix the underlying problem?

Reply:
Yes, I will do that and just remove the `defaultCheckTime` variable altogether. I think that would provide a much better experience for users.

Reply:
Or, actually, I do want to clarify. Are you OK with removing `defaultCheckTime`? This way, any extension which has not been updated yet will be updated immediately (well, after 10 seconds ( https://github.com/greasemonkey/greasemonkey/blob/b0f597970cd7205458675a83c2b6783ac34f9925/src/bg/updater.js#L129 )). This removes the requirement that the browser be running for 3 hours (which might especially be a stretch for Android) for any scripts to update when at least one has not been updated before.

Reply:
10 seconds after Firefox startup seems too short for this activity. In the past, there's been great concern from Mozilla about how long it takes Firefox to start up. I'd suggest a minimum value of at least several minutes (10? 15?) to assure that this additional processing, which isn't actually required for startup, is pushed well outside of the time during which the user is wanting to immediately interact with Firefox after startup.

Reply:
> 10 seconds after Firefox startup seems too short for this activity. In the past, there's been great concern from Mozilla about how long it takes Firefox to start up. I'd suggest a minimum value of at least several minutes (10? 15?) to assure that this additional processing, which isn't actually required for startup, is pushed well outside of the time during which the user is wanting to immediately interact with Firefox after startup.

Don’t worry. There already is a 15-minute pause because the first attempt at updates happens before the script registry loads, throwing an error which is caught at which point the first real automatic check is scheduled (I would like to make another separate change so that this is more explicit instead of accidental-looking—if I have the time):

https://github.com/greasemonkey/greasemonkey/blob/b0f597970cd7205458675a83c2b6783ac34f9925/src/bg/updater.js#L124

The part of logic that I am changing is recording the next attempt time on a script-by-script basis. If Firefox has been unopened for months, all scripts will have a next update time in the past. This use of the value of 0 for a never-before-update-checked script is identical to that scenario of scripts which have not been updated for months due to the browser not running for months.

Reply:
@makyen though, while this is off-topic for this particular PR, I personally would prefer that scripts which haven’t been updated for weeks are immediately updated upon opening Firefox. Since most scripts are updated due to website compatibility, staying out of date will provide a worse experience than any slowdown due to checking. Also, GreaseMonkey already serializes updates, so there is a limit to the performance impact.

Reply:
@binki Saying things like "this is off-topic, but I'm going to state my position anyway" is, generally, unhelpful to keeping a conversation on-topic, despite the fact that we all do it (including me), at least on occasion. By doing so, the other person is put in the position of either not responding and allowing the already expressed position be the only one stated there, which, at least to some readers, will be seen as implicit agreement, or to respond substantially to the off-topic comment and continue the off-topic conversation (which may or may not be appropriate, and likely disruptive to those who have chosen to get notifications of new content for whatever was on-topic).

"Should non-required things be done at startup or should they be delayed to later in order to make startup faster" is something that has had quite a bit of discussion over the years, particularly with respect to Firefox. Mozilla has come down *strongly* directing that everything that's not actually required should be delayed past startup, as Mozilla sees customer feedback of long startup times as one of the significant reasons users switch away from using Firefox.

While delaying non-required tasks is the correct choice in general, and it is something that extension writers should be significantly considering while writing their extension, if it's the right choice in any specific instance, is, often, debatable, and likely something that individual users would have differing opinions on.

Reply:
@makyen There, I opened #3193 for you.